src/main/java/com/meta/cp4m/llm/HuggingFaceLlamaPlugin.java

```diff
@@ -50,7 +50,7 @@ public T handle(ThreadState<T> threadState) throws IOException {
     Optional<String> prompt = promptCreator.createPrompt(threadState);
     if (prompt.isEmpty()) {
       return threadState.newMessageFromBot(
-          Instant.now(), "I'm sorry but that request was too long for me.");
+          Instant.now(), "I'm sorry but that request was too long for me.", threadState.tail());
```

Comment:
isn't the last message always the tail? seems like `ThreadState` should manage that logic

Reply:
Addressed this!